src/App.tsx

```diff
@@ -1,54 +1,128 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+function debounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+type Props = {
+  appliedQuary: string,
+}
+
+function getFilteredPeople(
+  people: Person[], { appliedQuary }: Props,
+): Person[] {
+  const copiedPeople = [...people];
+
+  if (appliedQuary) {
+    const normalizedQuary = appliedQuary.toLowerCase().trim();
+
+    return copiedPeople.filter(
+      person => person.name.toLowerCase().includes(normalizedQuary) || null,
+    );
+  }
+
+  return people;
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  // const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuary, setAppliedQuary] = useState('');
+  const [userId, setUserId] = useState('');
+  const [touched, setTouched] = useState(false);
```

Comment:
```suggestion
  const [isFocus, setIsFocus] = useState(false);
```

not critical but it seems more informative name